server/controllers/admin.py

```diff
@@ -1164,3 +1166,121 @@ def start_test_job(cid):
             current_course=current_course,
             form=form,
         )
+
+##########
+# Canvas #
+##########
+
+@admin.route('/course/<int:cid>/canvas/')
+@is_staff(course_arg='cid')
+def canvas_course(cid):
+    courses, current_course = get_courses(cid)
+    canvas_course = CanvasCourse.by_course_id(cid)
+    if not canvas_course:
+        return redirect(url_for('.edit_canvas_course', cid=cid))
+    external_names = {
+        a['id']: a['name'] for a in canvas_api.get_assignments(canvas_course)
+    }
+    return render_template(
+        'staff/canvas/index.html',
+        courses=courses,
+        current_course=current_course,
+        canvas_course=canvas_course,
+        external_names=external_names,
+    )
+
+@admin.route('/course/<int:cid>/canvas/edit/', methods=['GET', 'POST'])
+@is_staff(course_arg='cid')
+def edit_canvas_course(cid):
+    courses, current_course = get_courses(cid)
+    canvas_course = CanvasCourse.by_course_id(cid)
+    if not canvas_course:
+        canvas_course = CanvasCourse(course_id=cid)
+    form = forms.CanvasCourseForm()
+    if form.validate_on_submit():
+        form.populate_canvas_course(canvas_course)
```

Comment:
I would like to avoid leaking a user's personal access token. If a professor puts their access token - any staff who has that info - has professor level access to _other_ courses. 

Instead of populating the object - could we just set the value of the URL and leave the access token blank